repository-ydb-v2/src/main/java/tech/ydb/yoj/repository/ydb/merge/ByEntityYqlQueriesMerger.java

```diff
@@ -160,41 +162,43 @@ private MergingState doTransition(MergingState state, Statement.QueryType nextQu
         return nextState;
     }
 
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings({"rawtypes", "unchecked"})
     private static YdbRepository.Query convertInsertToUpsert(YdbRepository.Query<?> query) {
-        var type = getEntityClass(query);
-        var schema = EntitySchema.of(type);
-        var tableDescriptor = convertQueryToYqlStatement(query).getTableDescriptor();
-        var statement = new UpsertYqlStatement<>(tableDescriptor, schema);
-        return new YdbRepository.Query<>(statement, query.getValues().get(0));
+        YqlStatement srcStatement = convertQueryToYqlStatement(query);
+        EntitySchema schema = srcStatement.getInSchema();
+        TableDescriptor tableDescriptor = srcStatement.getTableDescriptor();
```

Comment:
we can't use `var` any more, because of wildcards... (wildcards in inferred `EntitySchema<?>` and `TableDescriptor<?>` types are **not** compatible, but `UpsertYqlStatement` requires the same parameter type for `EntitySchema<T>` and `TableDescriptor<T>`)

and the same is true for `convertInsertToDelete()` method